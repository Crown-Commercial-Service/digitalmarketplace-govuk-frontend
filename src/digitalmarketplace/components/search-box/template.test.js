/**
 * @jest-environment jsdom
 */

const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('search-box')

describe('Search box', () => {
  describe('by default', () => {
    it('matches existing snapshot', () => {
      const $ = render('search-box', examples.default)
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('search-box', examples.default, false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('has a label', async () => {
      const $ = render('search-box', examples.default, false, false, true)
      const $label = $('label.dm-search-box__label').text().trim()
      expect($label).toBe(examples.default.inputLabel)
    })

    it('has the correct id', async () => {
      const $ = render('search-box', examples.default, false, false, true)
      const $el = $('#' + examples.default.id)
      expect($el).toBeTruthy()
    })

    it('has the correct input id', async () => {
      const $ = render('search-box', examples.default, false, false, true)
      const $el = $('#' + examples.default.inputId)
      expect($el).toBeTruthy()
    })
  })

  describe('with a value', () => {
    it('matches existing snapshot', () => {
      const $ = render('search-box', examples['with value'])
      expect($.html()).toMatchSnapshot()
    })

    it('passes basic accessibility tests', async () => {
      const $ = render('search-box', examples['with value'], false, false, true)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('has a value', async () => {
      const $ = render('search-box', examples.default, false, false, true)
      const $inputText = $('input.dm-search-box__input[value="' + examples['with value'].value + '"]')
      expect($inputText).toBeTruthy()
    })
  })
})
