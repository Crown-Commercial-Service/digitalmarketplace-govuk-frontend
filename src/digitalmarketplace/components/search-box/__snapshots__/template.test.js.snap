// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search box by default matches existing snapshot 1`] = `
"<html><head></head><body><div id=\\"search-keywords\\" role=\\"search\\" aria-label=\\"keywords\\" data-module=\\"dm-search-box\\">
  <div class=\\"dm-search-box dm-search-box--on-white\\">
    <label for=\\"keywords\\" class=\\"dm-search-box__label\\">
      Keyword search
    </label>
    <div class=\\"dm-search-box__item-wrapper\\">
      <input type=\\"search\\" id=\\"keywords\\" name=\\"q\\" title=\\"Keyword search\\" class=\\"dm-search-box__item dm-search-box__input js-class-toggle\\">
      <div class=\\"dm-search-box__item dm-search-box__submit-wrapper\\">
        <button type=\\"submit\\" class=\\"dm-search-box__submit\\">Search</button>
      </div>
    </div>
  </div>
</div></body></html>"
`;

exports[`Search box with a value matches existing snapshot 1`] = `
"<html><head></head><body><div id=\\"search-keywords\\" role=\\"search\\" aria-label=\\"keywords\\" data-module=\\"dm-search-box\\">
  <div class=\\"dm-search-box dm-search-box--on-white\\">
    <label for=\\"keywords\\" class=\\"dm-search-box__label\\">
      Keyword search
    </label>
    <div class=\\"dm-search-box__item-wrapper\\">
      <input type=\\"search\\" value=\\"This is a search term\\" id=\\"keywords\\" name=\\"q\\" title=\\"Keyword search\\" class=\\"dm-search-box__item dm-search-box__input js-class-toggle\\">
      <div class=\\"dm-search-box__item dm-search-box__submit-wrapper\\">
        <button type=\\"submit\\" class=\\"dm-search-box__submit\\">Search</button>
      </div>
    </div>
  </div>
</div></body></html>"
`;
